sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.IoT.TimeSeriesInsights
+{
+    /// <summary>
+    /// Hierarchies client that can be used to perform operations such as creating, listing, replacing and deleting Time Series hierarchies.
+    /// </summary>
+    public class HierarchiesClient
+    {
+        private readonly TimeSeriesHierarchiesRestClient _hierarchiesRestClient;
+        private readonly ClientDiagnostics _clientDiagnostics;
+
+        /// <summary>
+        /// Initializes a new instance of ModelSettings. This constructor should only be used for mocking purposes.
+        /// </summary>
+        protected HierarchiesClient()
+        {
+        }
+
+        internal HierarchiesClient(TimeSeriesHierarchiesRestClient hierarchiesRestClient, ClientDiagnostics clientDiagnostics)
+        {
+            Argument.AssertNotNull(hierarchiesRestClient, nameof(hierarchiesRestClient));
+            Argument.AssertNotNull(clientDiagnostics, nameof(clientDiagnostics));
+
+            _hierarchiesRestClient = hierarchiesRestClient;
+            _clientDiagnostics = clientDiagnostics;
+        }
+
+        /// <summary>
+        /// Gets Time Series Insights hierarchies in pages asynchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The pageable list <see cref="AsyncPageable{TimeSeriesHierarchy}"/> of Time Series hierarchies with the http response.</returns>
+        public virtual AsyncPageable<TimeSeriesHierarchy> GetHierarchiesAsync(
```

Comment:
GetAsync as opposed to GetHierarchiesAsync.

Reply:
yes that's correct. Let me change that.